DNN Platform/Library/Services/Installer/Installers/ScriptInstaller.cs

```diff
@@ -120,44 +119,31 @@ protected ProviderConfiguration ProviderConfiguration
 
         /// -----------------------------------------------------------------------------
         /// <summary>
-        /// Gets the Pre-Upgrade Script (if present) - this script will always run before
+        /// Gets a list of Pre-Upgrade Scripts (if present) - these scripts will always run before
         /// any upgrade scripts but not upon initial installation.
         /// </summary>
-        /// <value>An InstallFile.</value>
+        /// <value>A list of <see cref="InstallFile"/> instances.</value>
         /// -----------------------------------------------------------------------------
-        protected InstallFile PreUpgradeScript
+        protected List<InstallFile> PreUpgradeScripts

```

Comment:
While we're defining new properties, let's [expose interfaces instead of concrete implementations](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1002)
```suggestion
        protected IList<InstallFile> PreUpgradeScripts
```